### PR TITLE
feat: enable voting from Safe

### LINKED ISSUE #263: feat: support voting with Safe for Snapshot spaces
We should support Safe voting, similar than on Snapshot.

### PR DESCRIPTION
**Pull request by ChaituVR**
### Summary
Closes #263 

- Updated manifest.json to include icon and imported manifest in index.html (used for safe app)
- New composable `useSafeWallet.ts` to check if wallet connected is Safe wallet or not, Also checks if safe wallet is using same network as space
- Show error on Proposal page if Safe wallet's network is different from safe network 
- Should be able to vote off-chain with Safe wallet (both from safe app and wallet connect)
- Should be able to vote on-chain with Safe wallet (both from safe app and wallet connect)
- If signature is `0x` UI will  show a notification that vote is pending and waiting for other signers

### How to test
- #### Add Snapshot as a custom Safe App (You can also connect with safe's wallet-connect)
  - Go to your Safe -> Apps -> My Custom Apps -> Click on "Add Custom App" button
  - Paste the snapshot UI url, for example https://bafybeihf65ituuynsylpwh7s4nj7txfhy27d7nlbi2jhg3wigze6rr2i2y.on.fleek.co/#/
  - You will see following App card if everything is correct
  - image 
  - Once you click on "Add" you will see that app is added to the list 
  - image
  - Navigate to the proposal you want to vote
 
- #### Choosing between on-chain and off-chain with Safe wallet:
  - By default safe wallet will use off-chain, to enable on-chain, refer to: https://docs.snapshot.org/user-guides/using-safe-multi-sig#asynchronous-signing

 
- #### Try to vote
- If you are connected to a safe that is on different network than space's network, then you will see following error
- image
- Try voting off-chain and on-chain
- Try voting with a Safe that require one confirmation
- Try voting with a Safe that require multiple confirmations

### To-Do
- [ ] Before merging remove this https://github.com/snapshot-labs/sx-monorepo/blob/enable-voting-from-safe/apps/ui/src/views/Home.vue#L42-L46

### CODE REVIEW

File: apps/ui/src/views/Home.vue [hunk 1]
```diff
@@ -39,6 +39,13 @@ const SOCIALS = [
       <div class="space-y-2">
         <div class="eyebrow">Learn more</div>
         <div class="space-y-2">
+          <!-- Remove this link after testing -->
+          <!-- We can only open home page on Safe, change link to your space if required -->
+          <div>
+            <a href="/#/s:thanku.eth">
+              Snapshot off-chain space <IH-arrow-sm-right class="inline-block -rotate-45" />
+            </a>
+          </div>
```
Comment (ChaituVR) [inline apps/ui/src/views/Home.vue:48]: This is just for testing. remove this before merging

File: apps/ui/src/composables/useActions.ts [hunk 2]
```diff
@@ -91,10 +91,11 @@ export function useActions() {
 
       console.log('Receipt', receipt);
 
+      if (envelope.signatureData.signature === '0x')
+        uiStore.addNotification('success', 'Your vote is pending! waiting for other signers');
```
[the remark this answers is not in the document]
Comment (ChaituVR) [reply apps/ui/src/composables/useActions.ts:95]: Yes but in this case, we still have to call `network.actions.send(envelope)` to send all data to the snapshot sequencer before showing this message

File: apps/ui/src/components/ProposalVote.vue [hunk 3]
```diff
@@ -55,6 +58,9 @@ const isSupported = computed(() => {
   <slot v-else-if="!isSupported" name="unsupported">
     Voting for this proposal is not supported
   </slot>
+  <slot v-else-if="isInvalidNetwork" name="wrong-safe-network">
+    Safe's network should be same as space's network
+  </slot>
```
[the remark this answers is not in the document]
Comment (Sekhmet) [reply apps/ui/src/components/ProposalVote.vue:64]: Let's limit it to offchain spaces for now.

File: apps/ui/src/views/Landing.vue [hunk 4]
```diff
@@ -39,6 +39,11 @@ const SOCIALS = [
       <div class="space-y-2">
         <div class="eyebrow">Learn more</div>
         <div class="space-y-2">
+          <div>
+            <a href="/#/explore">
+              Explore page <IH-arrow-sm-right class="inline-block -rotate-45" />
+            </a>
+          </div>
```
Comment (Sekhmet) [inline apps/ui/src/views/Landing.vue:46]: Unless we actually want it independently (@bonustrack) from this PR can we remove it?

I was testing on my own space so I had to navigate to my space via URL anyway
```
https://app.safe.global/apps/open?safe=oeth:0xa1919d8538F69412F2DD2A001Fd76E34307d3F79&appUrl=https%3A%2F%2Fbafybeielq6yozdxzk65irjrxutidakz74ja7dtw5txvu626y3cwzfgmasa.on.fleek.co%2F%23%2Fs%3A0cf5e.eth
```

Comment (bonustrack) [reply apps/ui/src/views/Landing.vue:46]: I can confirm we dont want this on the landing page

File: apps/ui/src/components/ProposalVote.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (ChaituVR) [reply apps/ui/src/components/ProposalVote.vue:64]: > Let's limit it to offchain spaces for now.

Done already

File: apps/ui/src/views/Landing.vue (hunk 4 above)
Comment (ChaituVR) [reply apps/ui/src/views/Landing.vue:46]: Weirdly, the Safe URL is not working in dev mode. but yeah I've added it to do our tests. will remove

**Review by Sekhmet — APPROVED** [review]
tACK

**Pull request merged**